### ISSUE TITLE
508-defect-2 [SCREENREADER, COGNITION]: Bold text SHOULD be made headings, in this instance

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#7770]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

```diff
!  Team affected: VSA Public Websites; Project: AMA Phase 2
```

**Feedback framework**
- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

Within the subway steps content, the text that is currently `<p> <strong>Words</strong> … </p>`, these **should** be headings, since the screen reader does not indicate they are headings, and it isn't clear when it is read out — it sounds like a sentence isn't finished.

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a screen reader user, I want to understand the hierarchy and content of the page content using heading levels to save time and aid comprehension.

## Environment

- Operating System: all
- Browser: all
- Screenreading device: any
- Server destination: staging & production

## Steps to Recreate

1. Enter [https://www.va.gov/decision-reviews/supplemental-claim/](https://www.va.gov/decision-reviews/supplemental-claim/) in browser
2. Start screenreading device of your choice
3. Navigate to the subway steps
4. Listen to how the content is read by the screen reader, without looking at the screen
5. Verify that it is confusing, sounds like sentences end abruptly.

## Possible Fixes (optional)

Recommend changing these to headings, of the appropriate level.

**Comment by kelsonic** [comment]
Could we add a link to the URL where this error is coming up? The preview link doesn't work anymore 🙂

**Comment by jenstrickland** [comment]
Oy! That issue is on Supplemental Claims. 

The spot check epic has all these issues noted in the comments. I did a search for bold text but nothing came up, then subway and this came up.

**Comment by kelsonic** [comment]
Ah perfect, thanks!

**Comment by kelsonic** [comment]
Because this isn't an app that's maintained in `vets-website`, I believe this is a Drupal task as I can't seem to find it in `vets-website`:

![image](https://user-images.githubusercontent.com/12773166/82691290-e13a5280-9c1a-11ea-959c-37105b7f7e29.png)

**Comment by jenstrickland** [comment]
I think this is something Laura and Randi could do. I think, but am not sure, it may be in the free text field?

**Comment by kelsonic** [comment]
I believe so as well! 🙂

**Comment by stephorkandatx** [comment]
@RLHecht similar to the other ticket we just covered- if we make modifications, let's record here and have Jen validate. If we investigate and determine to not make changes, we can use this ticket to alert Jen with our reasoning. Thank you!

**Comment by RLHecht** [comment]
@stephorkandatx When I changed the bold text in the subway map to an H4 or even H5 for accessibility, when you preview it, it renders as the same size as the subway map headers, even though they are H3. Kevin Walsh explained that it's a front end issue with the process list. So if we want to make this fix, I think there's a global fix that would need to be done first.

**Comment by stephorkandatx** [comment]
Thanks @RLHecht for the update!

@jenstrickland would this be an item to add to your larger spreadsheet, or handled differently?

**Comment by jenstrickland** [comment]
@stephorkandatx I'm not certain. I just added sitewide, CMS, and tools-fe to the labels. 

As I understand it, this is a CMS template issue, where we need additional styles added. Is that correct? Where I'm uncertain is I believe @kevwalsh said that Public Websites owns these templates. I'm not sure how the templates are revised.

**Comment by meganhkelley** [comment]
@cmaeng @kevwalsh @brianalloyd I'm not sure if this is still an issue but upon skimming the comment history, it looks like it would need to be addressed by one of your teams. I'm removing my team's label; please add it back and @ me if this is incorrect!

**Comment by brianalloyd** [comment]
@meganhkelley thanks for the update.  This ticket looks like it's been around for a while so I'll engage with our team to address the issue.  Thanks.

**Comment by RLHecht** [comment]
@brianalloyd This ticket would be a FE task that's a global fix to the templates.

**Comment by brianalloyd** [comment]
Thanks @RLHecht for reviewing, we'll place for FE action on this ticket.  Thanks.

**Comment by kelsonic** [comment]
Hey @RLHecht so I took a look at our `process.drupal.liquid` template and actually I believe this does indeed need to be updated in Drupal:

![image](https://user-images.githubusercontent.com/12773166/111517943-38b22d80-871b-11eb-8d45-6424f7fd536d.png)

An example page is http://preview-prod.vfs.va.gov/preview?nodeId=3011 and it looks like within the 4th content block (process) the:

```html
<p>
  <strong>
    some text
  </strong>
</p>
```

tags need to be updated to:

```html
<h4 class="vads-u-margin-bottom--0 vads-u-font-size--base">
  some text
</h4>
```

**Comment by RLHecht** [comment]
@kelsonic I just changed the header size on those three lines in the subway map to H4s, but they look the same size as the H3s that are at the top of the subway map. Any suggestions? 
<img width="755" alt="Screen Shot 2021-03-17 at 3 18 20 PM" src="https://user-images.githubusercontent.com/48728214/111525437-03fea000-8734-11eb-989e-f6bfc2a99d3c.png">

**Comment by kelsonic** [comment]
Awesome! Ya so each of those `h4` tags will need the classes: `class="vads-u-margin-bottom--0 vads-u-font-size--base"`

**Comment by kelsonic** [comment]
Do you have the ability to write raw html for those headers?

**Comment by RLHecht** [comment]
I can. Can you show me exactly how I need to add that html though? Is what you have above what goes in front of the text?

**Comment by RLHecht** [comment]
After putting in the source code, and saving, Drupal took out what Kelson put in minus the h4 tag so it's still showing up as the same size as the H3s in the process list. <img width="693" alt="Screen Shot 2021-03-18 at 1 17 00 PM" src="https://user-images.githubusercontent.com/48728214/111668529-3a97f180-87ec-11eb-8fb7-b2b1ffbbfb2a.png">

**Comment by kelsonic** [comment]
Oh no... right because the CMS is likely sanitizing the HTML on save, which would omit the `class` property... Well, I can't think of another maintainable solution to this atm, not sure what we want to do 😬 fysa @brianalloyd @ncksllvn

**Comment by brianalloyd** [comment]
Solid question @kelsonic.  @ncksllvn do you want to solution on this for a moment.  If a solution doesn't exist we can place in backlog and revisit.  Thanks

**Comment by ncksllvn** [comment]
> When I changed the bold text in the subway map to an H4 or even H5 for accessibility, when you preview it, it renders as the same size as the subway map headers, even though they are H3. Kevin Walsh explained that it's a front end issue with the process list. So if we want to make this fix, I think there's a global fix that would need to be done first.

This is exactly it. We need to adjust the FE CSS for this process list, which seems to be setting the font size for every heading which isn't ideal. I'll look into it.

**Comment by ncksllvn** [comment]
This was a design system bug. Opened a PR for it, https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/pull/597

**Comment by ncksllvn** [comment]
The work here has been merged and will go live this afternoon. 

@RLHecht the issue was that we had a weird style rule that caused all heading tags to render the same size inside of process lists. I got rid of that rule, so heading tags should be distinguishable visually now. In the screenshot below, I changed a `<strong>` tag into an `<h4>`  so you can see what I mean -

![image](https://user-images.githubusercontent.com/1915775/112333162-bbde0100-8c90-11eb-8627-9d91ca4b46e3.png)

**Comment by brianalloyd** [comment]
Thanks @jenstrickland, validated and closing in support.
